optuna/study.py

```diff
@@ -315,6 +326,15 @@ def optimize(
                     _iter = iter(int, 1)
 
                 with Parallel(n_jobs=n_jobs, prefer="threads") as parallel:
+                    if not isinstance(parallel._backend, joblib.parallel.ThreadingBackend) and \
+                       show_progress_bar:
+                        msg = (
+                            'Current progress bar does not work well with joblib for '
+                            'multi-processing'
+                        )
+                        warnings.warn(msg, UserWarning)
+                        _logger.warning(msg)
```

Comment:
How about just disabling the progress bar? I believe this'll result in an error otherwise as the tqdm progress bar cannot be pickled.